File: v7/src/stringToBinary.py
def binary_string_to_bytes(binary_str):
    # Teile den Binärstring in Blöcke von 8 Bits und konvertiere sie in Bytes
    byte_list = [binary_str[i:i+8] for i in range(0, len(binary_str), 8)]
    return bytes([int(byte, 2) for byte in byte_list])

def convert_file_to_binary(input_file, output_file):
    with open(input_file, 'r') as infile:
        binary_str = infile.read().strip()

    # Konvertiere den Binärstring in Bytes
    byte_data = binary_string_to_bytes(binary_str)

    # Schreibe die Bytes in die Binärdatei
    with open(output_file, 'wb') as outfile:
        outfile.write(byte_data)

# Beispielaufruf
input_file = '../arduinoLogs/arduinoIn.txt'  # Die Datei, die den Binärstring enthält
output_file = '../arduinoLogs/arduinoIn.bin'  # Die Datei, in die die Binärdaten geschrieben werden

convert_file_to_binary(input_file, output_file)

File: v7/src/receive/stringToBinary.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <string>
#include <bitset>

// Funktion, um einen Binärstring in Bytes zu konvertieren
std::vector<uint8_t> binaryStringToBytes(const std::string &binaryStr) {
    std::vector<uint8_t> byteList;
    for (size_t i = 0; i < binaryStr.size(); i += 8) {
        std::string byteString = binaryStr.substr(i, 8);
        uint8_t byte = std::bitset<8>(byteString).to_ulong();
        byteList.push_back(byte);
    }
    return byteList;
}

// Funktion, um eine Datei von Binärstring zu Binärdatei zu konvertieren
void convertFileToBinary(const std::string &inputFile, const std::string &outputFile) {
    std::ifstream infile(inputFile);
    if (!infile) {
        std::cerr << "Fehler beim Öffnen der Eingabedatei." << std::endl;
        return;
    }

    std::string binaryStr;
    infile >> binaryStr;
    infile.close();

    std::vector<uint8_t> byteData = binaryStringToBytes(binaryStr);

    std::ofstream outfile(outputFile, std::ios::binary);
    if (!outfile) {
        std::cerr << "Fehler beim Öffnen der Ausgabedatei." << std::endl;
        return;
    }
    outfile.write(reinterpret_cast<const char*>(byteData.data()), byteData.size());
    outfile.close();
}

// Beispielaufruf
int main() {
    std::string inputFile = "../arduinoLogs/arduinoIn.txt";  // Die Datei, die den Binärstring enthält
    std::string outputFile = "../arduinoLogs/arduinoIn.bin"; // Die Datei, in die die Binärdaten geschrieben werden

    convertFileToBinary(inputFile, outputFile);

    return 0;
}
